--- a/CMakeLists.txt	2019-03-04 08:31:47.503107677 +0100
+++ b/CMakeLists.txt	2019-03-04 08:33:30.598101614 +0100
@@ -834,8 +834,10 @@
 
 ######################################################################################
 
-if(DEFINED INSTALL_OPENCL_HEADERS)
+if(INSTALL_OPENCL_HEADERS)
   message(STATUS "Install POCL's OpenCL headers: ${INSTALL_OPENCL_HEADERS}")
+elseif(DEFINED INSTALL_OPENCL_HEADERS AND NOT INSTALL_OPENCL_HEADERS)
+  message(STATUS "Not install POCL's OpenCL headers: ${INSTALL_OPENCL_HEADERS}")
 else() # Undefined = auto -> check
   find_file(OPENCL_H opencl.h PATH_SUFFIXES CL OpenCL)
   if(OPENCL_H)
